application.py

```diff
@@ -30,18 +33,23 @@ def index():
 
 @app.route("/admin")
 def admin():
-    users = db.query(User).all()
+    users = db.query(User).order_by(User.created_at.desc()).all()
     return render_template('users.html',users=users)  
 
 @app.route('/register', methods=['GET', 'POST']) 
 def register():
-    if request.method == 'POST':
-        import time   
+    if request.method == 'POST':        
         username = request.form['username']
         password = request.form["password"]
-        user = User(username=username, password=password, created_at=time.strftime('%Y-%m-%d %H:%M:%S'))
-        db.add(user)
-        db.commit()
+        password = hashlib.md5(password.encode()).hexdigest()
+        
+        exists = db.query(User.id).filter_by(username=username).scalar()
```

Comment:
Can it be done by catching an exception when the insert fails? That way you will reduce one request to the database.

Reply:
It can be, will do that